x-pack/filebeat/input/httpjson/input.go

```diff
@@ -196,12 +203,13 @@ func run(ctx v2.Context, cfg config, pub inputcursor.Publisher, crsr *inputcurso
 
 	metrics := newInputMetrics(reg)
 
-	client, err := newHTTPClient(stdCtx, cfg, log, reg)
+	client, err := newHTTPClient(stdCtx, cfg, stat, log, reg)
 	if err != nil {
+		ctx.UpdateStatus(status.Failed, "failed to create HTTP client: "+err.Error())
 		return err
 	}
 
-	requestFactory, err := newRequestFactory(stdCtx, cfg, log, metrics, reg)
+	requestFactory, err := newRequestFactory(stdCtx, cfg, stat, log, metrics, reg)
 	if err != nil {
 		log.Errorf("Error while creating requestFactory: %v", err)
 		return err
```

Comment:
`stat.UpdateStatus(status.Failed, "failed to create requestFactory: "+err.Error())` makes sense here ?